src/Expander/Task.hs

```diff
@@ -36,18 +37,19 @@ data ExpanderTask
   | AwaitingDefn Var Ident Binding SplitCorePtr Ty SplitCorePtr Syntax
     -- ^ Waiting on var, binding, and definiens, destination, syntax to expand
   | AwaitingType SplitTypePtr [AfterTypeTask]
-  | ExpandDecl DeclPtr Scope Syntax DeclValidityPtr
-    -- ^ Where to put it, the scope, the decl, where to put the phase
-  | ExpandMoreDecls ModBodyPtr Scope Syntax DeclValidityPtr
-    -- Depends on the binding of the name(s) from the decl and the phase
+  | ExpandDeclForms DeclTreePtr ScopeSet DeclValidityPtr DeclValidityPtr Syntax
+    -- ^ The produced declaration tree, some already-introduced scopes which
+    -- the syntax can already see, some to-be-introduced scopes which the will
+    -- see, a destination for all the introduced scopes, including those by the
+    -- Syntax's remaining declaration forms.
```

Comment:
Is this comment too long?

Reply:
It might be a sign that there's enough arguments to warrant a record here, but I think it's fine. Might be nicer as a bulleted list.